cirq-core/cirq/circuits/circuit.py

```diff
@@ -1450,6 +1450,64 @@ def zip(
                 ) from ex
         return result
 
+    def ragged_concat(
+        *circuits: 'cirq.AbstractCircuit', align: Union['cirq.Alignment', str] = Alignment.LEFT
+    ) -> 'cirq.AbstractCircuit':
+        """Concatenates circuits while overlapping them if possible.
```

Comment:
Maybe mention "ragged" edges here, e.g.
```suggestion
        """Concatenates circuits, overlapping them if possible due to ragged edges.
```